common/oauth.go

```diff
@@ -8,6 +8,21 @@ import (
 	"golang.org/x/oauth2"
 )
 
+// TokenSourceContext is an interface that extends the oauth2.TokenSource interface
+// with a context. This is useful for token sources that need to be aware of the
+// context in which they are being called. The use of this interface is optional,
+// but if the token source conforms to it, then the context version of the Token
+// method will be called instead of the normal one.
+type TokenSourceContext interface {
+	oauth2.TokenSource
+
+	// TokenContext returns a token or an error.
+	// Token must be safe for concurrent use by multiple goroutines.
+	// The returned Token must not be modified.
+	// Similar semantics to oauth2.TokenSource, but with a context.
+	TokenContext(ctx context.Context) (*oauth2.Token, error)
```

Comment:
maybe `TokenFromContext`? Since the return value is a token.

Reply:
I was trying to stay faithful to the typical go naming convention. The other interface just has a function called "Token", and usually in go if you add something taking a context, you just do "NameContext". e.g. `net.Dialer.Dial` vs `net.Dialer.DialContext`.

Also it's not extracting the token from the context, per se. It's using the context to understand its environment better, but the token will still come either from the DB or from the token manager service.

I could maybe do `TokenWithContext` or something like that?